heksher/db_logic/setting.py

```diff
@@ -289,3 +290,29 @@ async def set_setting_type(self, setting_name: str, new_type: SettingType):
         """
         async with self.db_engine.begin() as conn:
             await conn.execute(settings.update().where(settings.c.name == setting_name).values(type=str(new_type)))
+
+    async def rename_setting(self, old_name: str, new_name: str):
+        """
+        Change a canonical setting name to another one, adding the old one as an alias
+        Args:
+            old_name: The name of the setting to rename
+            new_name: The new name to set for the setting
+        """
+        async with self.db_engine.begin() as conn:
+            # this should cascade through all other tables
+            await conn.execute(
+                settings.update()
+                .where(settings.c.name == old_name)
+                .values({"name": new_name})
+            )
+            # add the old name as an alias of the new one
+            await conn.execute(
+                insert(setting_aliases).values(
+                    [{'setting': new_name, 'alias': old_name}]
+                )
+            )
+            # if we renamed the alias to be canonical, we need to remove it from the aliases table
```

Comment:
do you mean that, in case that the new name is an old alias, you remove the old alias? if so good, but clarify this comment